drake/doc/clion.rst

```diff
@@ -249,109 +251,164 @@ able to go to ``VCS > Commit Changes`` and there should be no changes seen.
-done through the right-click context menu.
+Open the ``Edit Tool`` for external tools as outlined above and enter the
+following values for the fields:
+
+  :Name: ``Clang Format Full File``
+  :Description: ``Apply clang-format to the active file``
+  :Program: ``clang-format``
+  :Parameters: ``-i $FileName$``
+  :Working directory: ``$FileDir$``
+
+Leave the checkbox options in their default state.
+
+Clang format selected lines
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Open the ``Edit Tool`` for external tools as outlined above and enter the
+following values for the fields:
+
+  :Name: ``Clang Format Selected Lines``
+  :Description: ``Apply clang-format to the selected lines``
+  :Program: ``clang-format``
+  :Parameters: ``-lines $SelectionStartLine$:$SelectionEndLine$ -i $FileName$``
+  :Working directory: ``$FileDir$``
+
+Leave the checkbox options in their default state.
+
+Correct #include ordering
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Open the ``Edit Tool`` for external tools as outlined above and enter the
+following values for the fields:
+
+  :Name: ``Clang Format Include Ordering``
+  :Description: ``Runs the clang format for correcting includes on the current
+                  file``
+  :Program: ``bazel``
+  :Parameters: ``run //drake/tools:clang-format-includes -- $FilePath$``
+  :Working directory: ``$Projectpath$``
+
+Leave the checkbox options in their default state.
+
+.. _integrating_lint_tools_with_clion:
+
+"Linting" files
+---------------
+
+"Linting" refers to using tools to find aspects of code which don't conform
+to specified coding practices. You can apply Drake's linting tools in CLion to
+find such issues. We'll define two tools:
+
+- General linting (via cpplint) which captures most of the Drake style guide.
+- Drake extended linting which captures aspects of the Drake style guide _not_
+  captured by the general linting tool. This includes detecting out-of-order
+  ``#include`` directives.
+
+These tools produce reports. In some cases, the reports can be automatically
+converted into clickable links so that you can click on a messsage and be taken
+to the file and line indicated in the message. The configuration instructions 
+include the details of how to configure these clickable links.
+
+You can also set the general coding style for CLion through the following steps
+
+1. Go to ``File`` > ``Settings`` > ``Editor`` > ``Code Style``
+2. On the right panel, Go to ``Default Options`` > ``Right margin (columns)``: 
+   Set it to 80
+3. Go to ``File`` > ``Settings`` > ``Editor`` > ``Code Style`` > ``C/C++``
+4. On the right panel, choose ``Set from`` > ``Predefined Style`` > ``Google``
+
+Lint selected file for google style guide
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Open the ``Edit Tool`` for external tools as outlined above and enter the
+following values for the fields:
+
+  :Name: ``Cpplint File``
+  :Description: ``Apply cpplint to the current file``
+  :Program: ``bazel``
```

Comment:
@SeanCurtis-TRI i love the clickable links, etc, but this is a set backwards for one reason:  i often go through and cpplint on a few select files while I'm running a command line bazel test //... (just before a commit/push).  Long story short, I can't cpplint in clion anymore during a command line build.  is going through the bazel command required?